### PR TITLE
Improvements to runnable jar

### PR DESCRIPTION
**Pull request by NottyCode**
Need to make some improvements to the runnable jar support. Need to:

1) Actually output debug messages
2) Allow include options in any order
3) Add some better debug messages
4) Default to .jar extension when a runnable jar is requested.

Test Overview

The existing regression tests were run to ensure that this continued to function as before. The error messages were tested manually, as was the default to .jar extension.

The NLS changes were reviewed before Open Liberty existed.

### CODE REVIEW

**Comment by NottyCode** [comment]
@LibbyBot Could you take a moment away from Turkey, Yams and Pumpkin pie and #build this PR please?

**Comment by NottyCode** [comment]
@LibbyBot I think I fixed the bug and fixed it, can you please #build this again please?

File: dev/wlp.lib.extract/resources/wlp/lib/extract/SelfExtractMessages.nlsprops [hunk 1]
```diff
@@ -276,6 +276,11 @@ UNABLE_TO_FIND_TOOLS_JAR=The Liberty Java agent could not be attached because th
 
 UNABLE_TO_FIND_JAVA_AGENT=The Liberty Java agent could not be located. This server will run, but it cannot perform functions that require the Java agent, such as monitoring or tracing.
 
+RUN_IN_CHILD_JVM_DEBUG=WLP_JAR_DEBUG was set. The server will run in a separate Java virtual machine.
+RUN_IN_CHILD_JVM_JVM_OPTIONS=The jar contains jvm.options files. The server will run in a separate Java virtual machine.
+RUN_IN_CHILD_JVM_IBM_AGENT_ISSUE=An IBM SDK is being used without the -XX:+EnableHCR option. The server will run in a separate Java virtual machine.
+RUN_IN_CHILD_JVM_JRE=A Java runtime environment installation is being used. The server will run in a separate Java virtual machine.
```
Comment (sishida) [inline dev/wlp.lib.extract/resources/wlp/lib/extract/SelfExtractMessages.nlsprops]: How about..."You are using a runtime environment installation. The server now runs in a separate virtual machine."

File: dev/wlp.lib.extract/resources/wlp/lib/extract/SelfExtractMessages.nlsprops [hunk 2]
```diff
@@ -276,6 +276,11 @@ UNABLE_TO_FIND_TOOLS_JAR=The Liberty Java agent could not be attached because th
 
 UNABLE_TO_FIND_JAVA_AGENT=The Liberty Java agent could not be located. This server will run, but it cannot perform functions that require the Java agent, such as monitoring or tracing.
 
+RUN_IN_CHILD_JVM_DEBUG=WLP_JAR_DEBUG was set. The server will run in a separate Java virtual machine.
+RUN_IN_CHILD_JVM_JVM_OPTIONS=The jar contains jvm.options files. The server will run in a separate Java virtual machine.
+RUN_IN_CHILD_JVM_IBM_AGENT_ISSUE=An IBM SDK is being used without the -XX:+EnableHCR option. The server will run in a separate Java virtual machine.
```
Comment (sishida) [inline dev/wlp.lib.extract/resources/wlp/lib/extract/SelfExtractMessages.nlsprops]: How about..."You are using a software development kit (SDK) without the -XX:+EnableHCR option. The server now runs in a separate virtual machine."

File: dev/wlp.lib.extract/resources/wlp/lib/extract/SelfExtractMessages.nlsprops [hunk 3]
```diff
@@ -276,6 +276,11 @@ UNABLE_TO_FIND_TOOLS_JAR=The Liberty Java agent could not be attached because th
 
 UNABLE_TO_FIND_JAVA_AGENT=The Liberty Java agent could not be located. This server will run, but it cannot perform functions that require the Java agent, such as monitoring or tracing.
 
+RUN_IN_CHILD_JVM_DEBUG=WLP_JAR_DEBUG was set. The server will run in a separate Java virtual machine.
+RUN_IN_CHILD_JVM_JVM_OPTIONS=The jar contains jvm.options files. The server will run in a separate Java virtual machine.
```
Comment (sishida) [inline dev/wlp.lib.extract/resources/wlp/lib/extract/SelfExtractMessages.nlsprops]: How about..."The .jar file contains jvm.options files. The server now runs in a separate virtual machine."

File: dev/wlp.lib.extract/resources/wlp/lib/extract/SelfExtractMessages.nlsprops [hunk 4]
```diff
@@ -276,6 +276,11 @@ UNABLE_TO_FIND_TOOLS_JAR=The Liberty Java agent could not be attached because th
 
 UNABLE_TO_FIND_JAVA_AGENT=The Liberty Java agent could not be located. This server will run, but it cannot perform functions that require the Java agent, such as monitoring or tracing.
 
+RUN_IN_CHILD_JVM_DEBUG=WLP_JAR_DEBUG was set. The server will run in a separate Java virtual machine.
```
Comment (sishida) [inline dev/wlp.lib.extract/resources/wlp/lib/extract/SelfExtractMessages.nlsprops]: How about..."You set the WLP_JAR_DEBUG property. The server now runs in a separate virtual machine."

**Comment by sishida** [comment]
Hello! I've added some suggested changes to the new messages. Suggested changes involve using active (instead of passive) voice, using present tense, and omitting product names. Thank you!

**Comment by NottyCode** [comment]
The build failures are in JAX-RS and no code was changed that would affect JAX-RS applications.

**Comment by sishida** [comment]
Update: We added the "ID review not needed" label and removed the "ID Required" label. I was informed that the messages were reviewed in another pull request. :)

**Pull request merged**